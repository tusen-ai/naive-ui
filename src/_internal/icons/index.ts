export { default as AddIcon } from './Add'
export { default as ArrowBackIcon } from './ArrowBack'
export { default as ArrowDownIcon } from './ArrowDown'
export { default as ArrowUpIcon } from './ArrowUp'
export { default as AttachIcon } from './Attach'
export { default as BackwardIcon } from './Backward'
export { default as CancelIcon } from './Cancel'
export { default as CheckmarkIcon } from './Checkmark'
export { default as ChevronDownIcon } from './ChevronDown'
export { default as ChevronDownFilledIcon } from './ChevronDownFilled'
export { default as ChevronLeftIcon } from './ChevronLeft'
export { default as ChevronRightIcon } from './ChevronRight'
export { default as ClearIcon } from './Clear'
export { default as CloseIcon } from './Close'
export { default as DateIcon } from './Date'
export { default as DownloadIcon } from './Download'
export { default as EmptyIcon } from './Empty'
export { default as ErrorIcon } from './Error'
export { default as EyeIcon } from './Eye'
export { default as EyeOffIcon } from './EyeOff'
export { default as FastBackwardIcon } from './FastBackward'
export { default as FastForwardIcon } from './FastForward'
export { default as FileIcon } from './File'
export { default as FilterIcon } from './Filter'
export { default as ForwardIcon } from './Forward'
export { default as InfoIcon } from './Info'
export { default as MoreIcon } from './More'
export { default as PhotoIcon } from './Photo'
export { default as RemoveIcon } from './Remove'
export { default as ResizeSmallIcon } from './ResizeSmall'
export { default as RetryIcon } from './Retry'
export { default as RotateClockwiseIcon } from './RotateClockwise'
export { default as RotateCounterclockwiseIcon } from './RotateCounterclockwise'
export { default as SearchIcon } from './Search'
export { default as SuccessIcon } from './Success'
export { default as SwitcherIcon } from './Switcher'
export { default as TimeIcon } from './Time'
export { default as ToIcon } from './To'
export { default as TrashIcon } from './Trash'
export { default as WarningIcon } from './Warning'
export { default as ZoomInIcon } from './ZoomIn'
export { default as ZoomOutIcon } from './ZoomOut'
