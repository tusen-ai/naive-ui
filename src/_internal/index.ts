export { default as NIconSwitchTransition } from './icon-switch-transition'
export { default as NFadeInExpandTransition } from './fade-in-expand-transition'
export { NBaseClose } from './close'
export { NBaseIcon } from './icon'
export { default as NBaseFocusDetector } from './focus-detector'
export { default as NBaseLoading } from './loading'
export type { BaseLoadingExposedProps } from './loading'
export { default as NInternalSelectMenu } from './select-menu'
export type { InternalSelectMenuRef } from './select-menu'
export { default as NBaseWave } from './wave'
export type { BaseWaveRef } from './wave'
export { default as NBaseMenuMask } from './menu-mask'
export { default as NInternalSelection } from './selection'
export type { InternalSelectionInst } from './selection'
export { default as NBaseSlotMachine } from './slot-machine'
export { default as NBaseClear } from './clear'
export { default as NBaseSuffix } from './suffix'
export { NScrollbar, NxScrollbar } from './scrollbar'
export type { ScrollbarInst, ScrollbarProps } from './scrollbar'
