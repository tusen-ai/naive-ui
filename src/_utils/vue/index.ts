export { getSlot } from './get-slot'
export { getVNodeChildren } from './get-v-node-children'
export { keep } from './keep'
export { omit } from './omit'
export { flatten } from './flatten'
export { call } from './call'
export { keysOf } from './keysOf'
export { render } from './render'
export { getFirstSlotVNode } from './get-first-slot-vnode'
export { createDataKey } from './create-data-key'
export { createRefSetter } from './create-ref-setter'
export { createInjectionKey } from './create-injection-key'
export {
  resolveSlot,
  resolveWrappedSlot,
  resolveSlotWithProps,
  resolveWrappedSlotWithProps,
  isSlotEmpty
} from './resolve-slot'
export type { MaybeArray } from './call'
export { mergeEventHandlers } from './merge-handlers'
export { isNodeVShowFalse } from './is-node-v-show-false'
export { Wrapper } from './wrapper'
