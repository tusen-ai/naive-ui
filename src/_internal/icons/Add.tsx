import { defineComponent, h } from 'vue'

export default defineComponent({
  name: 'Add',
  render() {
    return (
      <svg
        width="512"
        height="512"
        viewBox="0 0 512 512"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 112V400M400 256H112"
          stroke="currentColor"
          stroke-width="32"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    )
  }
})
