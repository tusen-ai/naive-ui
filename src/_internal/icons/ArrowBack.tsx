import { defineComponent, h } from 'vue'

export default defineComponent({
  name: 'ArrowBack',
  render() {
    return (
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M0 0h24v24H0V0z" fill="none"></path>
        <path d="M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42-.39-.39-1.02-.39-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z"></path>
      </svg>
    )
  }
})
