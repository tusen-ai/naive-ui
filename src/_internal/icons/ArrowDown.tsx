import { defineComponent, h } from 'vue'

export default defineComponent({
  name: 'ArrowDown',
  render() {
    return (
      <svg viewBox="0 0 28 28" version="1.1" xmlns="http://www.w3.org/2000/svg">
        <g stroke="none" stroke-width="1" fill-rule="evenodd">
          <g fill-rule="nonzero">
            <path d="M23.7916,15.2664 C24.0788,14.9679 24.0696,14.4931 23.7711,14.206 C23.4726,13.9188 22.9978,13.928 22.7106,14.2265 L14.7511,22.5007 L14.7511,3.74792 C14.7511,3.33371 14.4153,2.99792 14.0011,2.99792 C13.5869,2.99792 13.2511,3.33371 13.2511,3.74793 L13.2511,22.4998 L5.29259,14.2265 C5.00543,13.928 4.53064,13.9188 4.23213,14.206 C3.93361,14.4931 3.9244,14.9679 4.21157,15.2664 L13.2809,24.6944 C13.6743,25.1034 14.3289,25.1034 14.7223,24.6944 L23.7916,15.2664 Z" />
          </g>
        </g>
      </svg>
    )
  }
})
