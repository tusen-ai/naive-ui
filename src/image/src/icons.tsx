import { h } from 'vue'

export const prevIcon = (
  <svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M6 5C5.75454 5 5.55039 5.17688 5.50806 5.41012L5.5 5.5V14.5C5.5 14.7761 5.72386 15 6 15C6.24546 15 6.44961 14.8231 6.49194 14.5899L6.5 14.5V5.5C6.5 5.22386 6.27614 5 6 5ZM13.8536 5.14645C13.68 4.97288 13.4106 4.9536 13.2157 5.08859L13.1464 5.14645L8.64645 9.64645C8.47288 9.82001 8.4536 10.0894 8.58859 10.2843L8.64645 10.3536L13.1464 14.8536C13.3417 15.0488 13.6583 15.0488 13.8536 14.8536C14.0271 14.68 14.0464 14.4106 13.9114 14.2157L13.8536 14.1464L9.70711 10L13.8536 5.85355C14.0488 5.65829 14.0488 5.34171 13.8536 5.14645Z"
      fill="currentColor"
    />
  </svg>
)

export const nextIcon = (
  <svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M13.5 5C13.7455 5 13.9496 5.17688 13.9919 5.41012L14 5.5V14.5C14 14.7761 13.7761 15 13.5 15C13.2545 15 13.0504 14.8231 13.0081 14.5899L13 14.5V5.5C13 5.22386 13.2239 5 13.5 5ZM5.64645 5.14645C5.82001 4.97288 6.08944 4.9536 6.28431 5.08859L6.35355 5.14645L10.8536 9.64645C11.0271 9.82001 11.0464 10.0894 10.9114 10.2843L10.8536 10.3536L6.35355 14.8536C6.15829 15.0488 5.84171 15.0488 5.64645 14.8536C5.47288 14.68 5.4536 14.4106 5.58859 14.2157L5.64645 14.1464L9.79289 10L5.64645 5.85355C5.45118 5.65829 5.45118 5.34171 5.64645 5.14645Z"
      fill="currentColor"
    />
  </svg>
)

export const closeIcon = (
  <svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M4.089 4.216l.057-.07a.5.5 0 0 1 .638-.057l.07.057L10 9.293l5.146-5.147a.5.5 0 0 1 .638-.057l.07.057a.5.5 0 0 1 .057.638l-.057.07L10.707 10l5.147 5.146a.5.5 0 0 1 .057.638l-.057.07a.5.5 0 0 1-.638.057l-.07-.057L10 10.707l-5.146 5.147a.5.5 0 0 1-.638.057l-.07-.057a.5.5 0 0 1-.057-.638l.057-.07L9.293 10L4.146 4.854a.5.5 0 0 1-.057-.638l.057-.07l-.057.07z"
      fill="currentColor"
    />
  </svg>
)
