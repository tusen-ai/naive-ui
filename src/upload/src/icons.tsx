import { h } from 'vue'

/**
 * Since image is too large compared with normal icons, we keep it inside upload
 * now.
 */

export const imageIcon = (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 28 28">
    <g fill="none">
      <path
        d="M21.75 3A3.25 3.25 0 0 1 25 6.25v15.5A3.25 3.25 0 0 1 21.75 25H6.25A3.25 3.25 0 0 1 3 21.75V6.25A3.25 3.25 0 0 1 6.25 3h15.5zm.583 20.4l-7.807-7.68a.75.75 0 0 0-.968-.07l-.084.07l-7.808 7.68c.183.065.38.1.584.1h15.5c.204 0 .4-.035.583-.1l-7.807-7.68l7.807 7.68zM21.75 4.5H6.25A1.75 1.75 0 0 0 4.5 6.25v15.5c0 .208.036.408.103.593l7.82-7.692a2.25 2.25 0 0 1 3.026-.117l.129.117l7.82 7.692c.066-.185.102-.385.102-.593V6.25a1.75 1.75 0 0 0-1.75-1.75zm-3.25 3a2.5 2.5 0 1 1 0 5a2.5 2.5 0 0 1 0-5zm0 1.5a1 1 0 1 0 0 2a1 1 0 0 0 0-2z"
        fill="currentColor"
      >
      </path>
    </g>
  </svg>
)

export const documentIcon = (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 28 28">
    <g fill="none">
      <path
        d="M6.4 2A2.4 2.4 0 0 0 4 4.4v19.2A2.4 2.4 0 0 0 6.4 26h15.2a2.4 2.4 0 0 0 2.4-2.4V11.578c0-.729-.29-1.428-.805-1.944l-6.931-6.931A2.4 2.4 0 0 0 14.567 2H6.4zm-.9 2.4a.9.9 0 0 1 .9-.9H14V10a2 2 0 0 0 2 2h6.5v11.6a.9.9 0 0 1-.9.9H6.4a.9.9 0 0 1-.9-.9V4.4zm16.44 6.1H16a.5.5 0 0 1-.5-.5V4.06l6.44 6.44z"
        fill="currentColor"
      >
      </path>
    </g>
  </svg>
)
