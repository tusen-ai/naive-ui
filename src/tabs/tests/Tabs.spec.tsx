import { mount } from '@vue/test-utils'
import { h } from 'vue'
import { NTabPane, NTabs } from '../index'
import { AddIcon } from '../../_internal/icons'
import { sleep } from 'seemly'

describe('n-tabs', () => {
  it('should work with import on demand', () => {
    mount(NTabs)
  })

  it('should work with callback types', () => {
    function onUpdateValue1 (name: number): void {}
    function onUpdateValue2 (name: string): void {}
    function onUpdateValue3 (name: number | string): void {}
    mount(NTabs, {
      props: {
        onUpdateValue: onUpdateValue1
      }
    })
    mount(NTabs, {
      props: {
        onUpdateValue: onUpdateValue2
      }
    })
    mount(NTabs, {
      props: {
        onUpdateValue: onUpdateValue3
      }
    })
  })

  it('should work with empty tab-pane', () => {
    mount(NTabs, {
      props: {
        defaultValue: 'a'
      },
      slots: {
        default: () =>
          h(NTabPane, {
            tab: 'a',
            name: 'a'
          })
      }
    })
  })

  it('should show AddIcon with `addable` `on-add` prop', async () => {
    const onAdd = vi.fn()
    const wrapper = mount(NTabs, {
      props: {
        type: 'card',
        addable: true,
        onAdd
      }
    })

    expect(wrapper.findComponent(AddIcon).exists()).toBe(true)
    const addIcon = wrapper.find('.n-tabs-tab--addable')
    await addIcon.trigger('click')
    expect(onAdd).toHaveBeenCalled()
  })

  it('should work with `justify-content` prop', async () => {
    const wrapper = mount(NTabs)

    await wrapper.setProps({ justifyContent: 'space-between' })
    expect(wrapper.find('.n-tabs-wrapper').attributes('style')).toContain(
      'justify-content: space-between;'
    )

    await wrapper.setProps({ justifyContent: 'space-around' })
    expect(wrapper.find('.n-tabs-wrapper').attributes('style')).toContain(
      'justify-content: space-around;'
    )

    await wrapper.setProps({ justifyContent: 'space-evenly' })
    expect(wrapper.find('.n-tabs-wrapper').attributes('style')).toContain(
      'justify-content: space-evenly;'
    )
  })

  it('should work with `closable` prop', async () => {
    const wrapper = mount(NTabs, {
      props: {
        type: 'card',
        defaultValue: '1'
      },
      slots: {
        default: () => [
          h(NTabPane, {
            tab: '1',
            name: '1'
          })
        ]
      }
    })
    expect(wrapper.find('.n-base-close').exists()).toBe(false)

    await wrapper.setProps({ closable: true })
    expect(wrapper.find('.n-base-close').exists()).toBe(true)
  })

  it('should work with `size` prop', async () => {
    const wrapper = mount(NTabs)

    await wrapper.setProps({ size: 'small' })
    expect(wrapper.find('.n-tabs').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ size: 'medium' })
    expect(wrapper.find('.n-tabs').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ size: 'large' })
    expect(wrapper.find('.n-tabs').attributes('style')).toMatchSnapshot()
  })

  it('should work with `tabs-padding` prop', async () => {
    const wrapper = mount(NTabs)

    expect(
      wrapper.find('.n-tabs-scroll-padding').attributes('style')
    ).toContain('width: 0px;')

    await wrapper.setProps({ tabsPadding: 100 })
    expect(
      wrapper.find('.n-tabs-scroll-padding').attributes('style')
    ).toContain('width: 100px;')
  })

  it('should work with `display-directive` prop', async () => {
    const displayDirectives: Array<'show' | 'if' | 'show:lazy'> = [
      'show',
      'if',
      'show:lazy'
    ]
    const wrapper = mount(NTabs, {
      props: { value: 'show' },
      slots: {
        default: () =>
          displayDirectives.map((directive) => (
            <NTabPane
              displayDirective={directive}
              tab={directive}
              name={directive}
            >
              {{
                default: () => (
                  <span class={`test-${directive.replace(':', '-')}`} />
                )
              }}
            </NTabPane>
          ))
      }
    })
    await wrapper.setProps({ value: 'if' })
    expect(wrapper.find('.test-show').exists()).toEqual(true)
    expect(wrapper.find('.test-if').exists()).toEqual(true)
    expect(wrapper.find('.test-show-lazy').exists()).toEqual(false)
    await wrapper.setProps({ value: 'show:lazy' })
    expect(wrapper.find('.test-show').exists()).toEqual(true)
    expect(wrapper.find('.test-if').exists()).toEqual(false)
    expect(wrapper.find('.test-show-lazy').exists()).toEqual(true)
    await wrapper.setProps({ value: 'show' })
    expect(wrapper.find('.test-show').exists()).toEqual(true)
    expect(wrapper.find('.test-if').exists()).toEqual(false)
    expect(wrapper.find('.test-show-lazy').exists()).toEqual(true)
  })

  it('should work with `on-before-leave` prop', async () => {
    const wrapper = mount(NTabs, {
      props: {
        type: 'card',
        defaultValue: '3',
        onBeforeLeave: (name: string) => {
          switch (name) {
            case '1':
              return false
            case '2':
              // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
              return new Promise((resolve) => {
                setTimeout(() => {
                  resolve(true)
                }, 1000)
              }) as Promise<boolean>
            default:
              return true
          }
        }
      },
      slots: {
        default: () => [
          h(NTabPane, {
            tab: '1',
            name: '1'
          }),
          h(NTabPane, {
            tab: '2',
            name: '2'
          }),
          h(NTabPane, {
            tab: '3',
            name: '3'
          })
        ]
      }
    })
    const tabs = wrapper.findAll('.n-tabs-tab')
    expect(tabs[2].classes()).toContain('n-tabs-tab--active')
    await tabs[0].trigger('click')
    expect(tabs[2].classes()).toContain('n-tabs-tab--active')
    await tabs[1].trigger('click')
    expect(tabs[2].classes()).toContain('n-tabs-tab--active')
    await sleep(1000)
    expect(tabs[1].classes()).toContain('n-tabs-tab--active')
  })

  it('should work with `pane-class` prop', () => {
    const wrapper = mount(NTabs, {
      props: {
        paneClass: 'test'
      },
      slots: {
        default: () =>
          h(
            NTabPane,
            {
              tab: 'Oasis',
              name: 'oasis'
            },
            'Wonderwall'
          )
      }
    })

    expect(wrapper.find('.n-tab-pane').classes('test')).toBe(true)
  })

  it('should work with `pane-style` prop', () => {
    const wrapper = mount(NTabs, {
      props: {
        paneStyle: {
          color: 'red'
        }
      },
      slots: {
        default: () =>
          h(
            NTabPane,
            {
              tab: 'Oasis',
              name: 'oasis'
            },
            'Wonderwall'
          )
      }
    })

    expect(wrapper.find('.n-tab-pane').attributes('style')).toBe('color: red;')
  })

  it('should work with `tab-style` prop', () => {
    const wrapper = mount(NTabs, {
      props: {
        tabStyle: {
          color: 'red'
        }
      },
      slots: {
        default: () =>
          h(
            NTabPane,
            {
              tab: 'Oasis',
              name: 'oasis'
            },
            'Wonderwall'
          )
      }
    })

    expect(wrapper.find('.n-tabs-tab').attributes('style')).toBe('color: red;')
  })

  it('should work with `type` prop', () => {
    ;(['bar', 'line', 'card', 'segment'] as const).forEach((type) => {
      const wrapper = mount(NTabs, {
        props: {
          type
        },
        slots: {
          default: () =>
            h(
              NTabPane,
              {
                tab: 'Oasis',
                name: 'oasis'
              },
              'Wonderwall'
            )
        }
      })

      expect(wrapper.find('.n-tabs').classes()).toContain(
        `n-tabs--${type}-type`
      )
      wrapper.unmount()
    })
  })

  it('should work with `on-close` prop', async () => {
    const onClose = vi.fn()
    const wrapper = mount(NTabs, {
      props: {
        type: 'card',
        defaultValue: '1',
        closable: true,
        onClose
      },
      slots: {
        default: () => [
          h(NTabPane, {
            tab: '1',
            name: '1'
          })
        ]
      }
    })

    const addIcon = wrapper.find('.n-base-close')
    await addIcon.trigger('click')
    expect(onClose).toHaveBeenCalled()
  })

  it('should work with `prefix` `suffix` slots', async () => {
    const wrapper = mount(NTabs, {
      props: {
        defaultValue: '1'
      },
      slots: {
        default: () => [
          h(NTabPane, {
            tab: '1',
            name: '1'
          })
        ],
        prefix: () => 'test-prefix',
        suffix: () => 'test-suffix'
      }
    })

    expect(wrapper.find('.n-tabs-nav__prefix').exists()).toBe(true)
    expect(wrapper.find('.n-tabs-nav__prefix').text()).toBe('test-prefix')
    expect(wrapper.find('.n-tabs-nav__suffix').exists()).toBe(true)
    expect(wrapper.find('.n-tabs-nav__suffix').text()).toBe('test-suffix')
  })
})
