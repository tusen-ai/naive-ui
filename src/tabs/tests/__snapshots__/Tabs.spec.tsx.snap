// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`n-tabs should work with \`size\` prop 1`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-segment: rgb(247, 247, 250); --n-bar-color: #18a058; --n-tab-font-size: 14px; --n-tab-text-color: rgb(31, 34, 37); --n-tab-text-color-active: #18a058; --n-tab-text-color-disabled: rgba(194, 194, 194, 1); --n-tab-text-color-hover: #18a058; --n-pane-text-color: rgb(51, 54, 57); --n-tab-border-color: rgb(239, 239, 245); --n-tab-border-radius: 3px; --n-close-size: 18px; --n-close-icon-size: 14px; --n-close-color-hover: rgba(0, 0, 0, .09); --n-close-color-pressed: rgba(0, 0, 0, .13); --n-close-border-radius: 3px; --n-close-icon-color: rgba(102, 102, 102, 1); --n-close-icon-color-hover: rgba(102, 102, 102, 1); --n-close-icon-color-pressed: rgba(102, 102, 102, 1); --n-tab-color: rgb(247, 247, 250); --n-tab-font-weight: 400; --n-tab-font-weight-active: 400; --n-tab-padding: 4px 0; --n-tab-padding-vertical: 6px 12px; --n-tab-gap: 36px; --n-tab-gap-vertical: 8px; --n-pane-padding-left: 0; --n-pane-padding-right: 0; --n-pane-padding-top: 8px; --n-pane-padding-bottom: 0; --n-font-weight-strong: 500; --n-tab-color-segment: #FFF;"`;

exports[`n-tabs should work with \`size\` prop 2`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-segment: rgb(247, 247, 250); --n-bar-color: #18a058; --n-tab-font-size: 14px; --n-tab-text-color: rgb(31, 34, 37); --n-tab-text-color-active: #18a058; --n-tab-text-color-disabled: rgba(194, 194, 194, 1); --n-tab-text-color-hover: #18a058; --n-pane-text-color: rgb(51, 54, 57); --n-tab-border-color: rgb(239, 239, 245); --n-tab-border-radius: 3px; --n-close-size: 18px; --n-close-icon-size: 14px; --n-close-color-hover: rgba(0, 0, 0, .09); --n-close-color-pressed: rgba(0, 0, 0, .13); --n-close-border-radius: 3px; --n-close-icon-color: rgba(102, 102, 102, 1); --n-close-icon-color-hover: rgba(102, 102, 102, 1); --n-close-icon-color-pressed: rgba(102, 102, 102, 1); --n-tab-color: rgb(247, 247, 250); --n-tab-font-weight: 400; --n-tab-font-weight-active: 400; --n-tab-padding: 6px 0; --n-tab-padding-vertical: 8px 16px; --n-tab-gap: 36px; --n-tab-gap-vertical: 8px; --n-pane-padding-left: 0; --n-pane-padding-right: 0; --n-pane-padding-top: 12px; --n-pane-padding-bottom: 0; --n-font-weight-strong: 500; --n-tab-color-segment: #FFF;"`;

exports[`n-tabs should work with \`size\` prop 3`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color-segment: rgb(247, 247, 250); --n-bar-color: #18a058; --n-tab-font-size: 16px; --n-tab-text-color: rgb(31, 34, 37); --n-tab-text-color-active: #18a058; --n-tab-text-color-disabled: rgba(194, 194, 194, 1); --n-tab-text-color-hover: #18a058; --n-pane-text-color: rgb(51, 54, 57); --n-tab-border-color: rgb(239, 239, 245); --n-tab-border-radius: 3px; --n-close-size: 18px; --n-close-icon-size: 14px; --n-close-color-hover: rgba(0, 0, 0, .09); --n-close-color-pressed: rgba(0, 0, 0, .13); --n-close-border-radius: 3px; --n-close-icon-color: rgba(102, 102, 102, 1); --n-close-icon-color-hover: rgba(102, 102, 102, 1); --n-close-icon-color-pressed: rgba(102, 102, 102, 1); --n-tab-color: rgb(247, 247, 250); --n-tab-font-weight: 400; --n-tab-font-weight-active: 400; --n-tab-padding: 10px 0; --n-tab-padding-vertical: 10px 20px; --n-tab-gap: 36px; --n-tab-gap-vertical: 8px; --n-pane-padding-left: 0; --n-pane-padding-right: 0; --n-pane-padding-top: 16px; --n-pane-padding-bottom: 0; --n-font-weight-strong: 500; --n-tab-color-segment: #FFF;"`;
